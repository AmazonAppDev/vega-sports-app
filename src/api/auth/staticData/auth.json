{
  "auth": {
    "id": "nmbsd-234sdc-678-jklsdf",
    "name": "Freida",
    "surname": "Gomam",
    "email": "fgomam@fakemail.com",
    "address": "Nowhere 5 USA",
    "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/5p.jpg",
    "profiles": [
      {
        "id": "bada55-1dl00k-4tm33",
        "displayName": "Freida",
        "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/6p.jpg"
      },
      {
        "id": "l00k-1nt0-s0uL",
        "displayName": "Jennifer",
        "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/7p.jpg"
      }
    ]
  }
}
