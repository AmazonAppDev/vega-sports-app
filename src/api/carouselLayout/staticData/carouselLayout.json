{
  "carousellayout": [
    {
      "id": "e77effd2-60fd-4a75-9921-ee1596c4d248",
      "carouselType": "hero",
      "carouselTitle": null,
      "endpoint": "suggestedforyou"
    },
    {
      "id": "fac0db1d-60ee-4ff9-9c3f-fc1ca24fa6be",
      "carouselType": "square",
      "carouselTitle": "Live TV: fresh news from events",
      "endpoint": "livestreams"
    },
    {
      "id": "cf5bafe0-023b-4ff9-9c3f-fc1ca24fa6be",
      "carouselType": "card",
      "carouselTitle": "Sport Teams Hall of Fame",
      "endpoint": "teams"
    },
    {
      "id": "068f3841-023b-4de8-85d1-f410d1407528",
      "carouselType": "square",
      "carouselTitle": "Documentaries on Sport Stars",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc13a87-af56-42ef-a66e-b882a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f1389-845c-476d-820b-1aa388e7bf1a4",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "5d21039f-3ba5-491b-b481-324ceb742ac71",
      "carouselType": "card",
      "carouselTitle": "Champions League",
      "endpoint": "documentaries"
    },
    {
      "id": "fac0db1d-60ee-4ff9-9c3f-fc15ca24fa6be",
      "carouselType": "square",
      "carouselTitle": "Live TV: fresh news from events",
      "endpoint": "livestreams"
    },
    {
      "id": "cf5bafe0-023b-4ff9-9c3f-6fc1ca24fa6be",
      "carouselType": "card",
      "carouselTitle": "Sport Teams Hall of Fame",
      "endpoint": "teams"
    },
    {
      "id": "06844f3841-023b-4de8-85d1-7f410d14aa07528",
      "carouselType": "square",
      "carouselTitle": "Documentaries on Sport Stars",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc1113a87-af56-42ef-a66e8-b8jj8a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f11389-845c-476d-8290b-1aajj88e7bf1a4",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "5d221039f-3ba5-491b-b4081-32cejjb742ac71",
      "carouselType": "card",
      "carouselTitle": "Champions League",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc132a87-af56-42ef-a66e-bgh882a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c9177f14389-845c-476d-820b-s",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "9fc166113a87-af56-42ef-a66e8-b8jj8a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f5511389-845c-476d-8290b-1aajj88e7bf1a4",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "5d22441039f-3ba5-491b-b4081-32cejjb742ac71",
      "carouselType": "card",
      "carouselTitle": "Champions League",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc13233a87-af56-42ef-a66e-bgh882a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f1422389-845c-476d-820b-s",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    }
  ]
}
