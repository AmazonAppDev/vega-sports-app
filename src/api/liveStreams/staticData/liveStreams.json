{
  "livestreams": [
    {
      "id": "50128bae-e954-4233-8e15-cd5867a31370",
      "stream_date": "2022-12-17T04:52:43Z",
      "title": "Live from Century's Football Match",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/11.jpg",
      "sport_type": "surfing"
    },
    {
      "id": "d9f25760-7263-426a-b4d7-a25e14a26f03",
      "stream_date": "2022-06-09T11:58:23Z",
      "title": "Formula 1 Discussion Panel",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/12.jpg",
      "sport_type": "soccer"
    },
    {
      "id": "73780c3d-2f09-4ea0-b9c0-50bdf2a9ba0a",
      "stream_date": "2022-10-07T23:10:08Z",
      "title": "Inside the Championship Studio",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/13.jpg",
      "sport_type": "padel"
    },
    {
      "id": "4448fb6d-f55c-4975-8b0c-14d5356d807b",
      "stream_date": "2022-01-25T06:12:26Z",
      "title": "Post-Match Analysis Desk",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/14.jpg",
      "sport_type": "surfing"
    },
    {
      "id": "997afe05-d0fa-44c2-b62e-555062807623",
      "stream_date": "2022-08-20T07:38:40Z",
      "title": "Voices from the Sidelines",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/15.jpg",
      "sport_type": "surfing"
    },
    {
      "id": "dad18f0f-d753-47b7-9393-e30630a394cd",
      "stream_date": "2022-08-27T15:58:41Z",
      "title": "Breaking News: Sports Edition",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/16.jpg",  
      "sport_type": "skiing"
    },
    {
      "id": "18813033-6113-4da8-ad75-2a209da6380f",
      "stream_date": "2024-10-17T20:05:31Z",
      "title": "Live Debate: Game Changers",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/17.jpg",
      "sport_type": "football"
    },
    {
      "id": "449e40c9-1b3a-4a33-931f-5172b5abaf39",
      "stream_date": "2022-08-11T07:26:11Z",
      "title": "Studio Insights: Race Day",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/18.jpg",
      "sport_type": "marathon"
    },
    {
      "id": "386e49d7-3ff1-476a-82b9-b9a61870b3b3",
      "stream_date": "2022-04-30T05:45:24Z",
      "title": "Reporter’s Corner: Match Highlights",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/19.jpg",
      "sport_type": "running"
    },
    {
      "id": "9fbaaba2-b6d0-442e-83fe-529212062a4c",
      "stream_date": "2022-08-04T05:43:00Z",
      "title": "On-Air with the Experts",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/20.jpg",
      "sport_type": "soccer"
    },
    {
      "id": "c4253727-d8fe-4b22-b792-e1c79ecaa5f4",
      "title": "In-Studio with Top Analysts",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/24.jpg",
      "sport_type": "boxing"
    },
    {
      "id": "f4a39998-4937-4bb7-82e7-90108a12a702",
      "stream_date": "",
      "title": "Game Day Reporter Spotlight",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/25.jpg",
      "sport_type": "football"
    },
    {
      "id": "39abb98c-922d-4f10-b205-4b95a4d4babb",
      "stream_date": null,
      "title": "Behind the Mic: Sports Talk",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/26.jpg",
      "sport_type": "biking"
    },
    {
      "id": "967d3e67-473d-4351-9c10-3727f55297ee",
      "stream_date": "2022-08-27T15:58:41Z",
      "title": "Breaking News: Sports Edition",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/27.jpg",
      "sport_type": "running"
    },
    {
      "id": "074ac572-c778-4402-bd68-61438f28e10f",
      "stream_date": "2024-10-17T20:05:31Z",
      "title": "Live Debate: Game Changers",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/28.jpg",
      "sport_type": "basketball"
    },
    {
      "id": "c9027268-5ef4-4663-b979-a8ae527373a1",
      "stream_date": "2022-08-11T07:26:11Z",
      "title": "Studio Insights: Race Day",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/29.jpg",
      "sport_type": "baseball"
    },
    {
      "id": "7e1a09a4-1066-43d3-9ee9-0eb89ade5c18",
      "stream_date": "2022-04-30T05:45:24Z",
      "title": "Reporter’s Corner: Match Highlights",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/30.jpg",
      "sport_type": "land hockey"
    },
    {
      "id": "1eeadc2b-4e6f-461a-8f93-825ef54b2f48",
      "stream_date": "2022-08-04T05:43:00Z",
      "title": "On-Air with the Experts",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/31.jpg",
      "sport_type": "tennis"
    },
    {
      "id": "144509c0-424f-40e3-ad26-f1ae885e8b79",
      "stream_date": "2022-08-04T05:43:00Z",
      "title": "Pre-Game Strategy Breakdown",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/32.jpg",
      "sport_type": "running"
    },
    {
      "id": "93e85028-3a15-44e1-b976-813711a6b938",
      "stream_date": "2022-08-04T05:43:00Z",
      "title": "Final Whistle Recap Show",
      "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/33.jpg",
      "sport_type": "biking"
    }
  ]
}