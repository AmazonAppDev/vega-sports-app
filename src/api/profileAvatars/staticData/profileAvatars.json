{
  "profileavatars": [
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/1p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/2p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/3p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/4p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/5p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/6p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/7p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/8p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/9p.jpg"
  ]
}
