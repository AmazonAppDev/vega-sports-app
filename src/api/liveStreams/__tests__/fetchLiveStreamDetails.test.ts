import { useQuery } from '@tanstack/react-query';
import { renderHook } from '@testing-library/react-native';

import { AppConfig } from '@AppServices/appConfig';
import {
  fetchLiveStreamDetailsApiCall,
  useLiveStreamDetails,
} from '../fetchLiveStreamDetails';
import staticData from '../staticData/liveStreamDetails.json';

jest.mock('@tanstack/react-query');

const mockRQ = useQuery as jest.Mock;

describe('fetchLiveStreamDetails', () => {
  beforeAll(() => {
    require('jest-fetch-mock').enableMocks();
  });

  afterEach(() => {
    fetchMock.resetMocks();
  });

  describe('fetchLiveStreamDetailsApiCall', () => {
    it('should return parsed livestream details data on success', async () => {
      fetchMock.mockResponseOnce(
        JSON.stringify({ data: staticData.livestreams }),
        {
          status: 200,
        },
      );

      const result = await fetchLiveStreamDetailsApiCall(
        '50128bae-e954-4233-8e15-cd5867a31370',
      );

      const { isOnAir, ...resultRest } = result ?? {};

      // compare all fields in object expect functions
      expect(resultRest).toMatchSnapshot();

      // expect isOnAir to be function
      expect(typeof isOnAir).toBe('function');
    });

    it('should return undefined for unexisting id', async () => {
      fetchMock.mockResponseOnce(
        JSON.stringify({ data: staticData.livestreams }),
        {
          status: 200,
        },
      );

      const result = await fetchLiveStreamDetailsApiCall('unexisting-id');

      expect(result).toBeUndefined();
    });

    test.each([{ itemId: '' }, { itemId: null }, { itemId: undefined }])(
      'should return undefined if livestream details data has nullable id ($itemId)',
      async ({ itemId }) => {
        // @ts-expect-error intentionally break TS
        await expect(fetchLiveStreamDetailsApiCall(itemId)).rejects.toThrow(
          'fetchLiveStreamDetailsApiCall() was used with invalid item id',
        );
      },
    );

    describe('isOnAir', () => {
      afterEach(() => {
        jest.runOnlyPendingTimers();
        jest.useRealTimers();
      });

      test.each([
        { now: '2022-12-17T04:00:00.000Z', isOnAirResult: false },
        { now: '2022-12-17T07:00:00.000Z', isOnAirResult: true },
        { now: '2022-12-17T14:47:00.000Z', isOnAirResult: true },
      ])(
        'should return $isOnAirResult if now date is $now',
        async ({ now, isOnAirResult }) => {
          jest.useFakeTimers().setSystemTime(new Date(now));

          fetchMock.mockResponseOnce(
            JSON.stringify({ data: staticData.livestreams }),
            {
              status: 200,
            },
          );

          const result = await fetchLiveStreamDetailsApiCall(
            '50128bae-e954-4233-8e15-cd5867a31370',
          );

          // expect isOnAir to be function
          expect(typeof result?.isOnAir).toBe('function');
          expect(result?.isOnAir?.()).toBe(isOnAirResult);
        },
      );

      test.each([
        {
          itemId: 'undefined-stream-date',
          streamDateValue: undefined,
          now: '2022-12-17T04:00:00.000Z',
        },
        {
          itemId: 'empty-stream-date',
          streamDateValue: '',
          now: '2022-12-17T07:00:00.000Z',
        },
        {
          itemId: 'null-stream-date',
          streamDateValue: null,
          now: '2022-12-17T14:47:00.000Z',
        },
      ])(
        'should return undefined if start or end date is nullable ($streamDateValue)',
        async ({ now, itemId }) => {
          jest.useFakeTimers().setSystemTime(new Date(now));

          fetchMock.mockResponseOnce(
            JSON.stringify({ data: staticData.livestreams }),
            {
              status: 200,
            },
          );

          const result = await fetchLiveStreamDetailsApiCall(itemId);

          // expect isOnAir to be function
          expect(typeof result?.isOnAir).toBe('function');
          expect(result?.isOnAir?.()).toBeUndefined();
        },
      );

      test.each([
        {
          itemId: 'null-stream-duration',
          streamDurationValue: null,
          now: '2022-12-17T04:00:00.000Z',
        },
        {
          itemId: 'no-stream-duration',
          streamDurationValue: undefined,
          now: '2022-12-17T07:00:00.000Z',
        },
        {
          itemId: 'zero-stream-duration',
          streamDurationValue: 0,
          now: '2022-12-17T14:47:00.000Z',
        },
      ])(
        'should return undefined if stream duration is nullable ($streamDurationValue)',
        async ({ now, itemId }) => {
          jest.useFakeTimers().setSystemTime(new Date(now));

          fetchMock.mockResponseOnce(
            JSON.stringify({ data: staticData.livestreams }),
            {
              status: 200,
            },
          );

          const result = await fetchLiveStreamDetailsApiCall(itemId);

          // expect isOnAir to be function
          expect(typeof result?.isOnAir).toBe('function');
          expect(result?.isOnAir?.()).toBeUndefined();
        },
      );
    });

    it('should throw error when API responded with 400 status', async () => {
      jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

      fetchMock.mockResponseOnce(JSON.stringify({ livestreams: {} }), {
        status: 400,
      });

      await expect(
        fetchLiveStreamDetailsApiCall('50128bae-e954-4233-8e15-cd5867a31370'),
      ).rejects.toThrow(
        `fetchLiveStreamDetailsApiCall(): resources does not exist for endpoint 'livestreams'`,
      );
    });

    it('should throw error when API responded with unexpected status', async () => {
      jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

      // @ts-expect-error intentionally break TS
      fetchMock.mockResponseOnce(null);

      await expect(
        fetchLiveStreamDetailsApiCall('50128bae-e954-4233-8e15-cd5867a31370'),
      ).rejects.toThrow(`Perform a request failed`);
    });

    test.each([{ status: 401 }, { status: 403 }, { status: 500 }])(
      'should throw error when API response with $status status',
      async ({ status }) => {
        jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

        fetchMock.mockResponseOnce(JSON.stringify({}), { status });

        await expect(
          fetchLiveStreamDetailsApiCall('50128bae-e954-4233-8e15-cd5867a31370'),
        ).rejects.toThrow(
          `fetchLiveStreamDetailsApiCall(): failed to fetch data from endpoint 'livestreams'`,
        );
      },
    );
  });

  describe('useLiveStreamDetails', () => {
    it('should return live streams data and no error on successful fetch', async () => {
      fetchMock.mockResponseOnce(
        JSON.stringify({ data: staticData.livestreams[0] }),
        {
          status: 200,
        },
      );

      mockRQ.mockReturnValueOnce({
        data: await fetchLiveStreamDetailsApiCall(
          '50128bae-e954-4233-8e15-cd5867a31370',
        ),
        error: null,
        isLoading: false,
      });

      const { result } = renderHook(() =>
        useLiveStreamDetails({
          liveStreamId: '50128bae-e954-4233-8e15-cd5867a31370',
        }),
      );

      const { isOnAir, ...resultRest } = result.current.data ?? {};

      // test hook state
      expect(result.current.error).toBe(null);
      expect(result.current.isLoading).toBe(false);

      // test hook return data shape
      expect(resultRest).toMatchSnapshot();

      // expect isOnAir to be function
      expect(typeof isOnAir).toBe('function');
    });

    describe('isOnAir', () => {
      afterEach(() => {
        jest.runOnlyPendingTimers();
        jest.useRealTimers();
      });

      test.each([
        { now: '2022-12-17T04:00:00.000Z', isOnAirResult: false },
        { now: '2022-12-17T07:00:00.000Z', isOnAirResult: true },
        { now: '2022-12-17T14:47:00.000Z', isOnAirResult: true },
      ])(
        'should return $isOnAirResult if now date is $now',
        async ({ now, isOnAirResult }) => {
          jest.useFakeTimers().setSystemTime(new Date(now));

          fetchMock.mockResponseOnce(
            JSON.stringify({ data: staticData.livestreams[0] }),
            {
              status: 200,
            },
          );

          mockRQ.mockReturnValueOnce({
            data: await fetchLiveStreamDetailsApiCall(
              '50128bae-e954-4233-8e15-cd5867a31370',
            ),
            error: null,
            isLoading: false,
          });

          const { result } = renderHook(() =>
            useLiveStreamDetails({
              liveStreamId: '50128bae-e954-4233-8e15-cd5867a31370',
            }),
          );

          // expect isOnAir to be function
          expect(typeof result.current.data?.isOnAir).toBe('function');
          expect(result.current.data?.isOnAir?.()).toBe(isOnAirResult);
        },
      );

      test.each([
        {
          itemId: 'undefined-stream-date',
          streamDateValue: undefined,
          now: '2022-12-17T04:00:00.000Z',
        },
        {
          itemId: 'empty-stream-date',
          streamDateValue: '',
          now: '2022-12-17T07:00:00.000Z',
        },
        {
          itemId: 'null-stream-date',
          streamDateValue: null,
          now: '2022-12-17T14:47:00.000Z',
        },
      ])(
        'should return undefined if start or end date is nullable ($streamDateValue)',
        async ({ now, itemId }) => {
          jest.useFakeTimers().setSystemTime(new Date(now));

          fetchMock.mockResponseOnce(
            JSON.stringify({ data: staticData.livestreams[0] }),
            {
              status: 200,
            },
          );

          mockRQ.mockReturnValueOnce({
            data: await fetchLiveStreamDetailsApiCall(itemId),
            error: null,
            isLoading: false,
          });

          const { result } = renderHook(() =>
            useLiveStreamDetails({ liveStreamId: itemId }),
          );

          // expect isOnAir to be function
          expect(typeof result.current.data?.isOnAir).toBe('function');
          expect(result.current.data?.isOnAir?.()).toBeUndefined();
        },
      );

      test.each([
        {
          itemId: 'null-stream-duration',
          streamDurationValue: null,
          now: '2022-12-17T04:00:00.000Z',
        },
        {
          itemId: 'no-stream-duration',
          streamDurationValue: undefined,
          now: '2022-12-17T07:00:00.000Z',
        },
        {
          itemId: 'zero-stream-duration',
          streamDurationValue: 0,
          now: '2022-12-17T14:47:00.000Z',
        },
      ])(
        'should return undefined if stream duration is nullable ($streamDurationValue)',
        async ({ now, itemId }) => {
          jest.useFakeTimers().setSystemTime(new Date(now));

          fetchMock.mockResponseOnce(
            JSON.stringify({ data: staticData.livestreams[0] }),
            {
              status: 200,
            },
          );

          mockRQ.mockReturnValueOnce({
            data: await fetchLiveStreamDetailsApiCall(itemId),
            error: null,
            isLoading: false,
          });

          const { result } = renderHook(() =>
            useLiveStreamDetails({ liveStreamId: itemId }),
          );

          // expect isOnAir to be function
          expect(typeof result.current.data?.isOnAir).toBe('function');
          expect(result?.current?.data?.isOnAir?.()).toBeUndefined();
        },
      );
    });

    it('should return isLoading as true before data loads', () => {
      mockRQ.mockReturnValueOnce({
        data: undefined,
        error: null,
        isLoading: true,
      });

      const { result } = renderHook(() =>
        useLiveStreamDetails({
          liveStreamId: '50128bae-e954-4233-8e15-cd5867a31370',
        }),
      );

      expect(result.current.isLoading).toBe(true);
      expect(result.current.data).toBeUndefined();
      expect(result.current.error).toBe(null);
    });

    test.each([
      '50128bae-e954-4233-8e15-cd5867a31370',
      '23232',
      'some string',
      undefined,
      null,
    ])('should return error (case: %s)', (inputValue) => {
      const mockError = new Error('Network error');

      mockRQ.mockReturnValueOnce({
        data: undefined,
        error: mockError,
        isLoading: false,
      });

      const { result } = renderHook(() =>
        // @ts-expect-error intentionally break TS
        useLiveStreamDetails({ liveStreamId: inputValue }),
      );

      expect(result.current.error).toBe(mockError);
      expect(result.current.data).toBe(undefined);
      expect(result.current.isLoading).toBe(false);
    });
  });
});
