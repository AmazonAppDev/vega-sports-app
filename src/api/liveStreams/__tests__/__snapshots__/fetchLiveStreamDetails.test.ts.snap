// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchLiveStreamDetails fetchLiveStreamDetailsApiCall should return parsed livestream details data on success 1`] = `
{
  "description": "Watch the action live! From breathtaking plays to game-changing moments, this stream has it all. Feel the intensity as the players give it their all, leaving everything on the field. Every pass and every goal could change the outcome of this thrilling match.

Stay tuned for every second of the excitement. The game is on, and the stakes couldn’t be higher. Whether it’s a last-minute goal or a stunning save, this is football at its finest. Don’t just watch—experience the passion, the drama, and the glory of the game.

Watch to join the live stream now and be part of the action! Amazing moments are waiting for you to grasp them!",
  "headline": "Football (Dragons vs Cougars)",
  "imageCover": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/900x700/30.jpg",
  "itemId": "50128bae-e954-4233-8e15-cd5867a31370",
  "streamDuration": 734,
  "streamEndDate": 2022-12-17T17:06:43.000Z,
  "streamStartDate": 2022-12-17T04:52:43.000Z,
  "team1": {
    "name": "Dragons",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/thumbnails/thumbnail-01.jpg",
  },
  "team2": {
    "name": "Cougars",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/thumbnails/thumbnail-02.jpg",
  },
  "title": "Basketball: Dragons | Cougars",
  "videoSource": {
    "format": "HLS",
    "thumbnailUrl": "https://d1v0fxmwkpxbrg.cloudfront.net/tears-of-steel/",
    "title": "Sample Stream",
    "type": "hls",
    "uri": "https://demo.unified-streaming.com/k8s/features/stable/video/tears-of-steel/tears-of-steel.ism/.m3u8",
  },
}
`;

exports[`fetchLiveStreamDetails useLiveStreamDetails should return live streams data and no error on successful fetch 1`] = `
{
  "description": "Watch the action live! From breathtaking plays to game-changing moments, this stream has it all. Feel the intensity as the players give it their all, leaving everything on the field. Every pass and every goal could change the outcome of this thrilling match.

Stay tuned for every second of the excitement. The game is on, and the stakes couldn’t be higher. Whether it’s a last-minute goal or a stunning save, this is football at its finest. Don’t just watch—experience the passion, the drama, and the glory of the game.

Watch to join the live stream now and be part of the action! Amazing moments are waiting for you to grasp them!",
  "headline": "Football (Dragons vs Cougars)",
  "imageCover": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/900x700/30.jpg",
  "itemId": "50128bae-e954-4233-8e15-cd5867a31370",
  "streamDuration": 734,
  "streamEndDate": 2022-12-17T17:06:43.000Z,
  "streamStartDate": 2022-12-17T04:52:43.000Z,
  "team1": {
    "name": "Dragons",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/thumbnails/thumbnail-01.jpg",
  },
  "team2": {
    "name": "Cougars",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/thumbnails/thumbnail-02.jpg",
  },
  "title": "Basketball: Dragons | Cougars",
  "videoSource": {
    "format": "HLS",
    "thumbnailUrl": "https://d1v0fxmwkpxbrg.cloudfront.net/tears-of-steel/",
    "title": "Sample Stream",
    "type": "hls",
    "uri": "https://demo.unified-streaming.com/k8s/features/stable/video/tears-of-steel/tears-of-steel.ism/.m3u8",
  },
}
`;
